File: Java/0728_Self_Dividing_Numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> result = new ArrayList<>();
        for(int i = left; i <= right; i++){
            if(check(i))
                result.add(i);
        }
        return result;
    }
    private boolean check(int i){
        boolean f = true;
        int d = i;
        while(d > 0){
            if(d%10 == 0 || i % (d%10) != 0){
                f = false;
                break;
            }
            d /= 10;
        }
        return f;
    }
}

File: CS/0728_Self_Dividing_Numbers.cs
public class Solution {
    public IList<int> SelfDividingNumbers(int left, int right) {
        IList<int> result = new List<int>();
        for(int i = left; i <= right; i++){
            if(check(i)){
                result.Add(i);
            }
        }
        return result;
    }
    private bool check(int n){
        int d = n;
        while(d > 0){
            if(d%10 != 0 && n % (d%10) == 0){
                d /= 10;
            }
            else{
                return false;
            }
        }
        return true;
    }
}
